python/src/ProjectionStrategyImplementation_doc.i.in

```diff
@@ -262,3 +262,34 @@ designProxy : :class:`~openturns.DesignProxy`
 %enddef
 %feature("docstring") OT::ProjectionStrategyImplementation::getDesignProxy
 OT_ProjectionStrategy_getDesignProxy_doc
+
+%define OT_ProjectionStrategy_isLeastSquares_doc
+"Get the least squares flag.
+
+There are two methods to compute the coefficients: integration
+or least squares.
+
+Returns
+-------
+isLeastSquares: bool
+    True if the coefficients are estimated from least squares."
+%enddef
+%feature("docstring") OT::ProjectionStrategyImplementation::isLeastSquares
+OT_ProjectionStrategy_isLeastSquares_doc
+
+// ---------------------------------------------------------------------
+
+%define OT_ProjectionStrategy_isModelSelection_doc
+"Get the model selection flag.
+
+A model selection method can be used to select the coefficients
+of the decomposition which enable to best predict the output.
+Model selection can lead to a sparse functional chaos expansion.
+
+Returns
+-------
+isModelSelection: bool
+    True if the method involves a model selection method."
```

Comment:
Do you really need to write this specific docstring every time? Is there no class inheritance involved?

Reply:
Yes, we have have to.

Reply:
Let me explain why this line is necessary. The actual doc of ProjectionStrategy comes from there:

https://github.com/openturns/openturns/blob/832d0bef23f657f31c845e39b105b978ff83f53f/python/src/ProjectionStrategy_doc.i.in#L33

Hence, the actual content is in ProjectionStrategyImplementation.

In order to be sure of this, I built the doc of ProjectionStrategy with this block.

![Pasted image 20240414225702](https://github.com/openturns/openturns/assets/31351465/3b9acbe8-8c9c-48c7-b496-7fa4fa636ae9)
**Figure 1.** The API doc of ProjectionStrategy when ProjectionStrategyImplementation documents isLeastSquares and involvesModelSelection.

Then I removed the API doc. code of these two methods and built again the doc. Here is the result.

![Pasted image 20240414230733](https://github.com/openturns/openturns/assets/31351465/d0ed538d-2155-46a7-87e9-a1359c402bef)
**Figure 2.** The API doc of ProjectionStrategy when ProjectionStrategyImplementation does *not* document isLeastSquares and involvesModelSelection.

I hope that this is convincing enough.

Reply:
It is, thanks!